### PR TITLE
Statemanager logging

### PR DESCRIPTION
**Pull request by Juliusan**
# Description of change

State manager logging refactored. It should log less trace type messages. Timer tick event is logged only if it performs something useful. Status is logged less often: ever 10s instead of every timer tick. I also fixed a minor issue with state manager not being able to retrieve required state from the store.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

All the regular tests (including all the cluster tests) pass.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the [contribution guidelines](https://wiki.iota.org/smart-contracts/contribute) for this project
- [x] I have performed a self-review of my own code
- [x] I have selected the `develop` branch as the target branch
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/chain/statemanager/state_manager.go [hunk 1]
```diff
@@ -281,6 +281,10 @@ func (smT *stateManager) run() { //nolint:gocyclo
 			} else {
 				timerTickCh = nil
 			}
+		case <-statusTimerCh:
+			statusTimerCh = smT.timers.TimeProvider.After(constStatusTimerTime)
+			smT.log.Debugf("State manager loop iteration; there are %v inputs, %v messages, %v public key changes waiting to be handled",
+				smT.inputPipe.Len(), smT.messagePipe.Len(), smT.nodePubKeysPipe.Len())
```
Comment (jorgemmsilva) [inline packages/chain/statemanager/state_manager.go:287]: is this really needed?

File: packages/chain/statemanager/smGPA/state_manager_gpa.go [hunk 2]
```diff
@@ -399,42 +368,41 @@ func (smT *stateManagerGPA) traceBlockChainWithCallback(lastCommitment *state.L1
...
+			return nil
 		}
 		return smT.traceBlockChain(newBlockFetcherWithRelatedFetcher(previousCommitment, fetcher))
 	}
-	result, err := smT.markFetched(fetcher)
-	smT.log.Debugf("Tracing block %s chain done, err=%v", commitment, err)
-	return result, err
+	result := smT.markFetched(fetcher)
+	smT.log.Debugf("Block %s is already committed", commitment)
+	return result
```
Comment (jorgemmsilva) [inline packages/chain/statemanager/smGPA/state_manager_gpa.go:400]: are tall these necessary? can't we just log on the "error/missing" cases?

File: packages/chain/statemanager/smGPA/state_manager_gpa.go [hunk 3]
```diff
@@ -282,61 +269,47 @@ func (smT *stateManagerGPA) handleChainFetchStateDiff(input *smInputs.ChainFetch
...
 	oldRequestCallback := newBlockRequestCallback(isValidFun, func() {
 		oldBlockRequestCompleted = true
+		smT.log.Debugf("Input mempool state request for state (index %v, %s): new block request completed",
+			input.GetNewStateIndex(), input.GetNewL1Commitment())
 		respondIfNeededFun()
 	})
 	newRequestCallback := newBlockRequestCallback(isValidFun, func() {
 		newBlockRequestCompleted = true
+		smT.log.Debugf("Input mempool state request for state (index %v, %s): old block request completed",
+			input.GetNewStateIndex(), input.GetNewL1Commitment())
```
Comment (jorgemmsilva) [inline packages/chain/statemanager/smGPA/state_manager_gpa.go:307]: are these necessary?

File: packages/chain/statemanager/smGPA/state_manager_gpa.go [hunk 4]
```diff
@@ -282,61 +269,47 @@ func (smT *stateManagerGPA) handleChainFetchStateDiff(input *smInputs.ChainFetch
...
 		newChainOfBlocks = lo.Reverse(newChainOfBlocks[:len(newChainOfBlocks)-1])
 		newState, err := smT.store.StateByTrieRoot(input.GetNewL1Commitment().TrieRoot())
 		if err != nil {
-			smT.log.Errorf("error obtaining state: %w", err)
+			smT.log.Errorf("Input mempool state request for state (index %v, %s): error obtaining state: %w",
+				input.GetNewStateIndex(), input.GetNewL1Commitment(), err)
 			return
 		}
 		input.Respond(smInputs.NewChainFetchStateDiffResults(newState, newChainOfBlocks, oldChainOfBlocks))
+		smT.log.Debugf("Input mempool state request for state (index %v, %s): responded to chain with requested state,"+
```
Comment (jorgemmsilva) [inline packages/chain/statemanager/smGPA/state_manager_gpa.go:288]: is this necessary?

File: packages/chain/statemanager/smGPA/state_manager_gpa.go [hunk 5]
```diff
@@ -371,25 +341,24 @@ func (smT *stateManagerGPA) getBlock(commitment *state.L1Commitment) state.Block
...
-		return nil, nil // No messages to send
+		return nil // No messages to send
 	}
 	if smT.blocksToFetch.addCallback(lastCommitment, callback) {
 		smT.log.Debugf("Tracing block %s chain: the block is already being fetched", lastCommitment)
-		return nil, nil
+		return nil
 	}
 	if smT.blocksFetched.addCallback(lastCommitment, callback) {
 		smT.log.Debugf("Tracing block %s chain: the block is already fetched, but cannot yet be committed", lastCommitment)
```
Comment (jorgemmsilva) [inline packages/chain/statemanager/smGPA/state_manager_gpa.go:360]: are these logs necessary?

**Review by jorgemmsilva — APPROVED** [review]
maybe we can log only the errors/unexpected events?

**Comment by kape1395** [comment]
@jorgemmsilva, nothing is necessary until you need to understand some strange case.
It will log much less now; maybe that's enough for this step.
Later we can review it if the practice shows something is too much.

**Pull request merged**